### PR TITLE
Improve submission retrieving from Kons

### PR DESCRIPTION
**Pull request by dungwinux**
### List of changes:
- Change fetch submissions routine to fetch after 30, 60, 90 seconds. After 1 minutes 45 seconds, if the submission is not updated, it will automatically turned into _rejected_.
- Only ask the respective judger handle that one submission. This will reduce bandwidth usage.
- Stop database from re-updating submission that was updated before.
- **[Breaking changes]** Default value for when submission is added to DB is null instead of "Pending".
- Re-try submission when restarting server feature is moved to other branch.

### CODE REVIEW

File: src/controller/submitCode.js [hunk 1]
```diff
@@ -115,19 +121,28 @@ async function sendCode(source_code_path, user_id, prob_name) {
...
-        setTimeout(() => reloadSubs(), 45000);
+        judger
+            .send(source_code_path, prob_name, sub_id)
+            .then((res) => {
+                if (res.status !== 200) throw Error("Cannot get from Kon");
+            })
+            .then(() => {
+                // Attempt & retry 3 times to get submission
+                for (let i = 1; i < 4; i++)
+                    setTimeout(reloadSub, i * 30000, judger);
```
Comment (natsukagami) [inline src/controller/submitCode.js]: Does this mean the `reloadSub` function is **always** called after 30, 60 and 90 seconds? Even if they succeed from the first time?

Comment (dungwinux) [reply src/controller/submitCode.js]: Yes, this is a trial to test if Wafter & Kon are able to hold. Even if they succeed before, submission's result won't be re-updated

**Comment by minhducsun2002** [comment]
Is this branch considered ready for production usage?

**Comment by dungwinux** [comment]
Still need intensive testing and enhancement in sumbitting routine

**Pull request merged**